src/core/batch.rkt

```diff
@@ -82,71 +82,173 @@
       (unmunge root)))
   exprs)
 
-(define (expand-taylor batch)
-  (define vars (batch-vars batch))
+(define (expand-taylor input-batch)
+  (define vars (batch-vars input-batch))
+  (define nodes (batch-nodes input-batch))
+  (define roots (batch-roots input-batch))
+
+  ; Hash to avoid duplications
   (define icache (reverse vars))
```

Comment:
Why reverse?

Reply:
Yeah, that's wrong

Reply:
Hmm, it needs to be reverse because later we reverse on `icache` to extract nodes. 
This line help `vars` to be in the right order at the end